client/src/templates/Challenges/components/scene/scene-helpers.test.ts

```diff
@@ -0,0 +1,341 @@
+          dialogue: {
+            text: 'Left aligned',
+            align: 'left'
+          }
+        },
+        {
+          character: 'Quincy',
+          startTime: 1,
+          dialogue: {
+            text: 'Right aligned',
+            align: 'right'
+          }
+        },
+        {
+          character: 'Naomi',
+          startTime: 2,
+          dialogue: {
+            text: 'Center aligned',
+            align: 'center'
+          }
+        }
+      ];
+      const result = buildTranscript(commands);
+      expect(result).toBe(
+        '\n<strong>Naomi</strong>: Left aligned\n' +
+          '\n<strong>Quincy</strong>: Right aligned\n' +
+          '\n<strong>Naomi</strong>: Center aligned\n'
+      );
+    });
+
+    it('should handle commands with additional properties', () => {
+      const commands: SceneCommand[] = [
+        {
+          character: 'Naomi',
+          startTime: 0,
+          background: 'office.png',
+          position: { x: 100, y: 200 },
+          opacity: 0.8,
+          dialogue: {
+            text: 'Hello',
+            align: 'left'
+          }
+        },
+        {
+          character: 'Quincy',
+          startTime: 1,
+          finishTime: 5,
+          dialogue: {
+            text: 'Hi',
+            align: 'right'
+          }
+        }
+      ];
+      const result = buildTranscript(commands);
+      expect(result).toBe(
+        '\n<strong>Naomi</strong>: Hello\n' + '\n<strong>Quincy</strong>: Hi\n'
+      );
+    });
+
+    it('should preserve order of commands', () => {
+      const commands: SceneCommand[] = [
+        {
+          character: 'Naomi',
+          startTime: 2,
+          dialogue: {
+            text: 'Third',
+            align: 'left'
+          }
+        },
+        {
+          character: 'Quincy',
+          startTime: 0,
+          dialogue: {
+            text: 'First',
+            align: 'left'
+          }
+        },
+        {
+          character: 'Naomi',
+          startTime: 1,
+          dialogue: {
+            text: 'Second',
+            align: 'left'
+          }
+        }
+      ];
+      const result = buildTranscript(commands);
+      expect(result).toBe(
+        '\n<strong>Naomi</strong>: Third\n' +
+          '\n<strong>Quincy</strong>: First\n' +
+          '\n<strong>Naomi</strong>: Second\n'
+      );
+    });
+
+    it('should handle long dialogue text', () => {
+      const longText =
+        'This is a very long dialogue that spans multiple words and contains a lot of content that might be used in a scene.';
+      const commands: SceneCommand[] = [
+        {
+          character: 'Naomi',
+          startTime: 0,
+          dialogue: {
+            text: longText,
+            align: 'left'
+          }
+        }
+      ];
+      const result = buildTranscript(commands);
+      expect(result).toBe(`\n<strong>Naomi</strong>: ${longText}\n`);
+    });
+
+    it('should handle dialogue with newlines and whitespace', () => {
```

Comment:
We should check the newline case, but we don't need to check the space case.

`Hello      world` is essentially the as `Hello world`, which we already have in other test cases.